pilz_control/test/unittest_pilz_joint_trajectory_controller.cpp

```diff
@@ -434,127 +686,104 @@ TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testHoldAtStart)
+}
 
+TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testNotExecutingAfterUnhold)
+{
+  ASSERT_TRUE(performFullControllerStartup());
+  
+  bool is_executing_result;
   EXPECT_TRUE(invokeIsExecuting(is_executing_result));
-  EXPECT_FALSE(is_executing_result) << "Controller is executing unexpectedly";
+  EXPECT_FALSE(is_executing_result) << "There should be no execution to detect";
 }
 
-/**
- * @brief Test if the controller starts and completes the execution of a single action goal.
- */
-TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testSingleActionGoal)
+TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testActionGoalExecution)
 {
   ASSERT_TRUE(performFullControllerStartup());
 
-  /**********
-   * Step 1 *
-   **********/
-  ros::Duration goal_duration{2.0};
-  GoalType goal {generateSimpleGoal(goal_duration)};
-
+  GoalType goal {generateSimpleGoal()};
   trajectory_action_client_.sendGoal(goal);
+
+  std::chrono::milliseconds movement_timeout {WAIT_FOR_MOVEMENT_STARTED_MSEC};
+  EXPECT_TRUE(updateControllerUntil([this](){ return robot_->isMoving(); }, movement_timeout));
+
   bool is_executing_result;
-  EXPECT_TRUE(waitForIsExecutingResult(true, is_executing_result));
+  EXPECT_TRUE(invokeIsExecuting(is_executing_result));
+  EXPECT_TRUE(is_executing_result) << "Failed to detect action goal execution";
 
-  /**********
-   * Step 2 *
-   **********/
   ros::Duration default_update_period {DEFAULT_UPDATE_PERIOD_SEC};
-  progressInTime(goal_duration + default_update_period);
+  progressInTime(getGoalDuration(goal) + default_update_period);
   updateController();
 
   EXPECT_TRUE(invokeIsExecuting(is_executing_result));
-  EXPECT_FALSE(is_executing_result) << "Controller is executing unexpectedly";
+  EXPECT_FALSE(is_executing_result) << "There should be no execution to detect";
 }
 
-/**
- * @brief Test if the controller starts and completes the execution of a single trajectory command.
- */
-TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testSingleCommandMessage)
+TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testTrajCommandExecution)
 {
   ASSERT_TRUE(performFullControllerStartup());
 
-  /**********
-   * Step 1 *
-   **********/
-  ros::Duration goal_duration{2.0};
-  GoalType goal {generateSimpleGoal(goal_duration)};
-
+  GoalType goal {generateSimpleGoal()};
   trajectory_command_publisher_->publish(goal.trajectory);
+
+  std::chrono::milliseconds movement_timeout {WAIT_FOR_MOVEMENT_STARTED_MSEC};
+  EXPECT_TRUE(updateControllerUntil([this](){ return robot_->isMoving(); }, movement_timeout));
+
   bool is_executing_result;
-  EXPECT_TRUE(waitForIsExecutingResult(true, is_executing_result));
+  EXPECT_TRUE(invokeIsExecuting(is_executing_result));
+  EXPECT_TRUE(is_executing_result) << "Failed to detect trajectory command execution";
 
-  /**********
-   * Step 2 *
-   **********/
   ros::Duration default_update_period {DEFAULT_UPDATE_PERIOD_SEC};
```

Comment:
I think this line is obsolete and actually clutters the test code. Simply integrate it directly in the line where the variable is used.

Reply:
Done.